Dashboard/ViewModels/DataViewModel.cs

```diff
@@ -38,12 +37,10 @@ private void OnDataChanged(object? sender, EventArgs e)
     /// <summary>
     ///     Handles the cleanup when the view model is no longer needed.
     /// </summary>
-    public void Dispose()
+    public override void Dispose()
```

Comment:
My IDE doesn't like this change, by overriding we are no longer implementing the IDisposable interface correctly.

Also, there needs to be some more looking into this -- I added the disposable interface / pattern so that when the application closes the data store and its connector would also stop, it was running on a separate thread so it wouldn't just stop without being told to do so. So, maybe we need to work out how to make it stop all the connectors etc just when the app stops, instead of implementing it via IDisposable

Reply:
I forgot to remove the IDisposable interface from DataViewModel, if we have ViewModelBase implement IDispoable Dispose and then the rest of the ViewModels override ViewModelBases implementation is that not fine?

Reply:
19b1f940a5ff9e1743d7213780f9944f81b4fd27

Reply:
My main concern with disposing of these models, is we will have to reconstruct our connector (and data store) every time we open a new page. Is there an easier way to run code when the window exits? If the main window is always present, we can just go through our services and dispose of them in that on close method?

Reply:
I think we leave the dispose in the MainWindowViewModel, but instead we just fetch the datastore and call dispose on it. When the main window is closed, then everything else can be disposed

Reply:
Ah, I understand ya. I've adjusted it to fetch the datastore and dispose.
https://github.com/nickowen44/AV-Datalogger/pull/6/commits/02c2535675153fa40236d8366181b54e3d67d42e